File: depend/libcmb/perl/test3.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Cmb;

my @items = qw/a b c d/;
my $choose = 2;
my $nitems = $#items + 1;

printf STDERR "Enumerating choose-%s from %u:\n", $choose, $nitems;
my $cmb = new Cmb { size_min => $choose, size_max => $choose };
$cmb->cmb_callback($nitems, \@items, sub { printf "\t%s\n", "@_" });

File: depend/libcmb/python/test3.py
#!/usr/bin/env python
from cmb import *
from sys import stderr

items = ["a", "b", "c", "d"]
choose = 2
nitems = len(items)

print >> stderr, "Enumerating choose-%s from %u:" % (choose, nitems)
config = CMB(size_min = choose, size_max = choose)
def afunc(config, seq, nitems, items):
    print "\t%s" % " ".join(items)
    return 0
cmb_callback(config, nitems, items, afunc)
